airflow/providers/amazon/aws/hooks/rds.py

```diff
@@ -48,3 +50,291 @@ class RdsHook(AwsGenericHook["RDSClient"]):
     def __init__(self, *args, **kwargs) -> None:
         kwargs["client_type"] = "rds"
         super().__init__(*args, **kwargs)
+
+    def get_db_snapshot_state(self, snapshot_id: str) -> str:
+        """
+        Get the current state of a DB instance snapshot.
+
+        :param snapshot_id: The ID of the target DB instance snapshot
+        :return: Returns the status of the DB snapshot as a string (eg. "available")
+        :rtype: str
+        :raises AirflowNotFoundException: If the DB instance snapshot does not exist.
+        """
+        try:
+            response = self.conn.describe_db_snapshots(DBSnapshotIdentifier=snapshot_id)
+        except self.conn.exceptions.DBSnapshotNotFoundFault as e:
+            raise AirflowNotFoundException(e)
```

Comment:
I wrapped the relevant "resource missing" exceptions in `AirflowNotFoundException` rather than propagating them to the caller. This makes it easier for users to catch "resource doesn't exist" exceptions.